osrparse/replay.py

```diff
@@ -350,7 +350,7 @@ class Replay:
     rng_seed: Optional[int]
 
     @staticmethod
-    def from_path(path):
+    def from_path(path, use_legacy_parser=False, header_only=False):
```

Comment:
```suggestion
    def from_path(path, *, use_legacy_parser=False, header_only=False):
```

similarly for other static parsing methods.

Also unsure about exposing the legacy parsing at all. I think I'd prefer if we were entirely backed by osucore so we don 't have to maintain two versions.